storage/replica_raftstorage.go

```diff
@@ -560,16 +560,17 @@ func (r *Replica) applySnapshot(batch engine.Engine, snap raftpb.Snapshot) (uint
 		// the snapshot.
 		atomic.StoreUint64(&r.appliedIndex, snap.Metadata.Index)
 
-		// Atomically update the descriptor and lease.
-		if err := r.setDesc(&desc); err != nil {
-			panic(err)
-		}
 		// Update other fields which are uninitialized or need updating.
 		// This may not happen if the system config has not yet been loaded.
 		// While config update will correctly set the fields, there is no order
 		// guarangee in ApplySnapshot.
 		// TODO: should go through the standard store lock when adding a replica.
-		if err := r.updateRangeInfo(); err != nil {
+		if err := r.updateRangeInfo(&desc); err != nil {
+			panic(err)
+		}
+
+		// Atomically update the descriptor and lease.
```

Comment:
This comment is out of date. Replace it with "Update the range descriptor. This is done last as this is the step that makes the Replica visible in the Store."